langchain4j-azure-open-ai/src/test/java/dev/langchain4j/model/azure/AzureOpenAiAiServicesWithJsonSchemaIT.java

```diff
@@ -0,0 +1,41 @@
+package dev.langchain4j.model.azure;
+
+import dev.langchain4j.model.chat.ChatLanguageModel;
+import dev.langchain4j.service.AiServicesWithJsonSchemaIT;
+import org.junit.jupiter.api.AfterEach;
+
+import java.util.List;
+
+import static dev.langchain4j.model.chat.request.ResponseFormat.JSON;
+
+class AzureOpenAiAiServicesWithJsonSchemaIT extends AiServicesWithJsonSchemaIT {
+
+    @Override
+    protected List<ChatLanguageModel> models() {
+        return List.of(
+                AzureOpenAiChatModel.builder()
+                        .endpoint(System.getenv("AZURE_OPENAI_ENDPOINT"))
+                        .apiKey(System.getenv("AZURE_OPENAI_KEY"))
+                        .deploymentName("gpt-4o-mini")
+                        .responseFormat(JSON)
```

Comment:
`json_object` and `json_schema` are different features and we should distinguish between them as not all models support `json_schema`. Because of this, user has to opt-in in order to use `json_schema` and `strict`.
Current users already do `.responseFormat(new ChatCompletionsJsonResponseFormat())` to opt-in into `json_object`.
We should have a similar alternative for `json_schema`.

Reply:
If you select JSON, you can have "strict" which means you use the schema, or not "strict" that makes you use the `json_object`

Reply:
AFAIU `strict` does not mean "there is a schema", it means "adhere to the schema".
There are the following options/combinations:
1. no response format -> text response
2. `ChatCompletionsTextResponseFormat` -> text response
3. `ChatCompletionsJsonResponseFormat` -> JSON response without schema, requires describing schema in a free-form text in the prompt
4. `ChatCompletionsJsonSchemaResponseFormat` + `strict = true` -> JSON response with schema, guarantees schema adherance. This will require processing the schema (on the first request only) which takes around 10 seconds.
5. `ChatCompletionsJsonSchemaResponseFormat` + `strict = false` -> JSON response with schema, does not guarantee schema adherance. This does not require processing the schema.

So there is currently no way to distinguish between 3 and 5.
3 is used by many users for sure, as it is currently the only way to ensure json output. So we should be backwads compatible and still support it.
5 could be a "corner case" when schema is generated dynamically and/or provided by the end users and 10 second latency is not acceptable. It would be great to support this use case as well.

Would it be possible to use Azure's `ChatCompletionsResponseFormat` type instead of LC4j's `ResponseFormatType`?

Examples of how users would configure it:
3.
```java
AzureOpenAiChatModel.builder()
    .endpoint(System.getenv("AZURE_OPENAI_ENDPOINT"))
    .apiKey(System.getenv("AZURE_OPENAI_KEY"))
    .deploymentName("gpt-4o-mini")
    .responseFormat(new ChatCompletionsJsonResponseFormat())
    .build()
```

4.
```java
AzureOpenAiChatModel.builder()
    .endpoint(System.getenv("AZURE_OPENAI_ENDPOINT"))
    .apiKey(System.getenv("AZURE_OPENAI_KEY"))
    .deploymentName("gpt-4o-mini")
    .responseFormat(new ChatCompletionsJsonSchemaResponseFormat(null)) // not sure about null
    .strictJsonSchema(true)
    .build()
```

5.
```java
AzureOpenAiChatModel.builder()
    .endpoint(System.getenv("AZURE_OPENAI_ENDPOINT"))
    .apiKey(System.getenv("AZURE_OPENAI_KEY"))
    .deploymentName("gpt-4o-mini")
    .responseFormat(new ChatCompletionsJsonSchemaResponseFormat(null)) // not sure about null
    .strictJsonSchema(false) // or not set at all, false by default
    .build()
```

Another option would be to use OpenAI's strings `json_object` and `json_schema` in ctor/builder.

A bit of a background info: for vanilla OpenAI there is no way to set the actual schema (complete structure with fields) in the constructor/builder, only to "enable `json_schema` support", as currently not all models support this feature, only latest `gpt-4o/-mini` ones. When `ChatLanguageModel` is used with AI Service and return type is a POJO, we want to use Structured Outputs feature (with json schema), but only if `RESPONSE_FORMAT_JSON_SCHEMA` capability is enabled explicitly. If this capability is not enabled, AI Service will fall back to generating pseudo schema and append it to the user message. If we'll enable `RESPONSE_FORMAT_JSON_SCHEMA` for `OpenAiChatModel`/`AzureOpenAiChatModel` by default, this will lead to errors if users use older models. So right now to use Structured Outputs feature, users will need to _**opt in**_. I guess we will need to rethink and change this soon (no later than right before 1.0), as more and more users will migrate to newer models. I can imagine in 1 year older models will be decomissioned and newer will all support this feature, so there will be no need to require users to opt in, we could just enbale it by default. If user does not want to use it, we could add an option to opt out.

WDYT?